tt_metal/impl/buffers/buffer_constants.hpp

```diff
@@ -16,6 +16,7 @@ enum class TensorMemoryLayout {
     HEIGHT_SHARDED,
     WIDTH_SHARDED,
     BLOCK_SHARDED,
+    NONE
```

Comment:
I think this is a bad idea, and way too broad of a change without thinking about it a great deal. A better way would be to pass in an optional of TensorMemoryLayout and not have a value for the desired scenario.

Reply:
Yeah I though about it, I'm open for suggestion if people think optional is better I'll do that.
My preference is NONE.

Reply:
I think making it an optional param for conv is better for now. Exposing this in TensorMemoryLayout means users can pass this to any of our apis which may not be ready/expect this, and having an explicit NONE layout doesn't really make sense to me.